net_udp.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procfs
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type (
+	// NetUDPLine is a line parsed from /proc/net/udp
+	// For the proc file format details, see https://linux.die.net/man/5/proc
+	NetUDPLine struct {
+		TxQueue uint64
+		RxQueue uint64
+	}
+
+	NetUDP struct {
+		TxQueueLength uint64
+		RxQueueLength uint64
+		UsedSockets   uint64
+	}
+)
+
+// NetUDP returns kernel/networking statistics for udp datagrams read from /proc/net/udp.
+func (fs FS) NetUDP() (*NetUDP, error) {
```

Comment:
Since this is a more specialty parser than the typical procfs reader, I might suggest we have a name like `NetUDPSummary()`, or `NetUDPStats()`. What do you think @pgier?

Reply:
If I am allowed to add my 2 cents: The name should represents the location of the file in the proc filesystem. If someone else needs to extract some other data from this file, she/he/it could extend the function/struct instead of creating another file/reader. From my perspective, there is also not much more you can get from the content of the lines.

Reply:
Most of our other parsers dump the raw contents of the file into a struct, rather than parse and produce some derived metrics. I agree, that the function should be named after the filepath, but since this isn't a raw dump, we should either have it be a different function name, or have a struct method to summarize.

So, the NetUDP type struct is a full contents of the file, and then you would call `netUdp.Summary()` to get TxQueueLength, etc.

Reply:
+1. We should provide Go representations of the raw data and allow the caller to determine how the data is used. The convenience method could be reasonable if the logic is nontrivial.